ContextDesktop/src/main/java/com/github/migi_1/Context/main/HUDController.java

```diff
@@ -54,6 +82,21 @@ public void updateHUD() {
             float height = settings.getHeight();
             hudText.setLocalTranslation(width, height, 0);
         }
+        Vector3f commanderLoc = main.getEnv().getCommander().getModel().getLocalTranslation();
+        if (commanderLoc.x % CHECKPOINT_DISTANCE > -15 && commanderLoc.x < 0) {
+            main.getGuiNode().attachChild(checkpointAlertText);
+            if (checkpointUpdated) {
+                main.getEnv().setDifficulty(main.getEnv().getDifficulty() * DIFFICULTY_INCREASE);
+                checkpointUpdated = false;
+            }
+        } else {
+            if (!checkpointUpdated) {
+                checkpointCounter += 1;
+                checkpointAlertText.setText("CHECKPOINT " + Integer.toString(checkpointCounter) + " REACHED");
+                checkpointUpdated = true;
+            }
+            main.getGuiNode().detachChild(checkpointAlertText);
+        }
```

Comment:
Also, some comments as to what happens in this code would be useful.
For example ` if (commanderLoc.x % CHECKPOINT_DISTANCE > -15 && commanderLoc.x < 0) {` I can't quite make out here what's supposed to happen.